model/Image.php

```diff
@@ -1037,8 +1033,9 @@ class Image_Cached extends Image {
 	 * @param boolean $isSingleton This this to true if this is a singleton() object, a stub for calling methods.
 	 *                             Singletons don't have their defaults set.
 	 */
-	public function __construct($filename = null, $isSingleton = false) {
+	public function __construct($filename = null, $isSingleton = false, Image $sourceImage = null) {
 		parent::__construct(array(), $isSingleton);
+		if ($sourceImage) $this->update($sourceImage->toMap());
```

Comment:
In hindsight should I have done this?
```php
$properties = $sourceImage->toMap();
unset($properties['RecordClassName'], $properties['ClassName']);
if ($sourceImage) $this->update($properties);
```
is it hazardous to pass the RecordClassName and ClassName properties?